controllers/users-controllers.js

```diff
@@ -0,0 +1,69 @@
+const bcrypt = require('bcryptjs');
+const jwt = require('jsonwebtoken');
+require('dotenv').config();
+const User = require('../repositories/users-repository');
+const isUniqueUser = require('../helpers/is-unique');
+const { HttpCodes } = require('../helpers/constants');
+
+const createUser = async (req, res, next) => {
+  try {
+    const { name, email, password } = req.body;
+
+    const isUnique = await isUniqueUser(email);
+
+    if (!isUnique) {
+      return res
+        .status(HttpCodes.CONFLICT)
+        .json({ message: 'This email is already in use.' });
+    }
+
+    const hashedPassword = await bcrypt.hash(password, 8);
+
+    const addedUser = await User.addNewUser({
+      name,
+      email,
+      password: hashedPassword,
+    });
+
+    return res
+      .status(HttpCodes.CREATED)
+      .json({ message: 'You have successfully registered.', user: addedUser });
+  } catch (error) {
+    next(error);
+  }
+};
+
+const loginUser = async (req, res, next) => {
+  try {
+    const { email, password } = req.body;
+
+    const user = await User.getUserByEmail(email);
+
+    if (!user) {
+      return res
+        .status(HttpCodes.UNAUTHORIZED)
+        .json({ message: 'Invalid credentials.' });
+    }
+
+    const isPasswordCorrect = await bcrypt.compare(password, user.password);
```

Comment:
I suggest to move password encryption and comparing to the separate service.

Reply:
Done :)